Deferred/FutureCollections.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  FutureCollections.swift
+//  Deferred
+//
+//  Created by Zachary Waldowski on 8/29/15.
+//  Copyright © 2014-2015 Big Nerd Ranch. Licensed under MIT.
+//
+
+import Dispatch
+
+public extension SequenceType where Generator.Element: FutureType {
+    /// Choose the future that is determined first from a collection of futures.
+    ///
+    /// - returns: A deferred value that is determined with the first of the
+    ///   given futures to be determined.
+    var firstFuture: Deferred<Generator.Element.Value> {
```

Comment:
Might be better to use "earliest" rather than "first", since that gets the time aspect in there directly, while the latter is more indirect "if we map time onto a timeline, then the earliest one is the first one as you move forward".